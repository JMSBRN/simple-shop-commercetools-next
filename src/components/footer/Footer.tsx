import Categories from '../categories/Categories';
import MainLogo from '../main-logo/MainLogo';
import styles from './Footer.module.scss';
import { useTranslation } from 'next-i18next';
import SocialLinks from '../social-links/SocialLinks';

const Footer = () => {
  const { footer, container, section } = styles;
  const { t } = useTranslation('common');

  return (
    <footer data-testid="footer" className={footer}>
      <div className={container}>
        <div className={section}>
          <MainLogo />
          <p>
            {t('shopSlogan')}
          </p>
        </div>
        <div className={section}>
          <h3>{t('explore')}</h3>
          <Categories />
        </div>
        <div className={section}>
          <h3>{t('contactUsMessage')}</h3>
           <SocialLinks />
        </div>
      </div>
    </footer>
  );
};

export default Footer;
