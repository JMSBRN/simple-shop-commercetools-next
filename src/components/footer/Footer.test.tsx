import { mockUseRouter, renderWithProviders } from '@/tests/utilsForTests';
import Footer from './Footer';
import { CommonType } from 'types';

jest.mock('react-i18next', () => ({
    useTranslation: () => ({ t: (key: CommonType) => key }),
}));

mockUseRouter();
const renderFooter = () => {
    const { getByTestId, asFragment } = renderWithProviders(<Footer />);
    return {
        getByTestId,
        asFragment
    }
};

describe('Footer Component', () => {
    test('renders Footer component', () => {
        const { getByTestId } = renderFooter();
        expect(getByTestId('footer')).toBeInTheDocument();
    });

    test('renders MainLogo component inside Footer', () => {
        const { getByTestId } = renderFooter();
        expect(getByTestId('main-logo')).toBeInTheDocument();
    });

    test('renders Categories component inside Footer', () => {
        const { getByTestId } = renderFooter();
        expect(getByTestId('categories')).toBeInTheDocument();
    });

    test('renders SocialLinks component inside Footer', () => {
        const { getByTestId } = renderFooter();
        expect(getByTestId('social-links')).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { asFragment } = renderFooter();
        expect(asFragment()).toMatchSnapshot();
    });
});
