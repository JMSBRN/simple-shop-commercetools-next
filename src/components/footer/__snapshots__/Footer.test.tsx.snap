// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
    data-testid="footer"
  >
    <div
      class="container"
    >
      <div
        class="section"
      >
        <div
          class="logoStyle"
          data-testid="main-logo"
        >
          Store
          <div>
            international
          </div>
        </div>
        <p>
          shopSlogan
        </p>
      </div>
      <div
        class="section"
      >
        <h3>
          explore
        </h3>
        <div
          class="categoriesContainer"
          data-testid="categories"
        />
      </div>
      <div
        class="section"
      >
        <h3>
          contactUsMessage
        </h3>
        <div
          class="socialLinks"
          data-testid="social-links"
        >
          <div>
            <a
              href="/"
              target="blank"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="facebook"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </div>
          <div>
            <a
              href="/"
              target="blank"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="twitter"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </div>
          <div>
            <a
              href="/"
              target="blank"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="instagram"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </div>
          <div>
            <a
              href="/"
              target="blank"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="linkedin"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </div>
          <div>
            <a
              href="/"
              target="blank"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="telegram"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
